app/services/settings/streaming/stream-settings.ts

```diff
@@ -112,6 +112,9 @@ export class StreamSettingsService extends PersistentStatefulService<IStreamSett
 
         if (parameter.name === 'streamType' && patch.streamType !== void 0) {
           parameter.value = patch.streamType;
+          // we should immediately save the streamType in OBS if it's changed
+          // otherwise OBS will not save 'key' and 'server' values
+          this.settingsService.setSettings('Stream', streamFormData);
```

Comment:
Yeah I found this too on my restream branch.  I just worked around it by calling `setSettings` twice.